WordPress/Classes/ViewRelated/Reader/ReaderCardsStreamViewController.swift

```diff
@@ -99,7 +104,13 @@ class ReaderCardsStreamViewController: ReaderStreamViewController {
         page = 1
         refreshCount += 1
 
-        cardsService.fetch(isFirstPage: true, refreshCount: refreshCount, success: success, failure: failure)
+        cardsService.fetch(isFirstPage: true, refreshCount: refreshCount, success: { [weak self] cardsCount, hasMore in
+            self?.trackApiResponse()
+            success(cardsCount, hasMore)
```

Comment:
👋 @leandroalonso !

The track event name is named `READER_DISCOVER_CONTENT_PRESENTED` but the way you are using the tracking suggests that you want to track the API response, **NOT** the `trackContentPresented`.

With that in mind, and in order to align our tracking logic with how the tracking is called maybe we want to do something like the below:

```
        cardsService.fetch(isFirstPage: true, refreshCount: refreshCount, success: { [weak self] cardsCount, hasMore in
            success(cardsCount, hasMore)
            self?.trackContentPresented()
        }, failure: { [weak self] error in
            failure(error)
            self?.trackContentPresented()
        })
```

Here, I am assuming that the `success(...)` and `failure(...)` is doing the UI update. Is my assumption correct? If it is correct, maybe we should consider the suggestion above.

Reply:
Yup! You're right, the title should be "track content being presented".

We're doing this track to see if the API being slow is making the user leave Discover. I guess we can keep it as it is (even tho because I find it a little bit weird to track API responses) also the iOS event is already being used in funnels.

Reply:
Thanks for the reply Leandro! 🌟 

Actually, the title is one of my concerns, my main concern is the order, triggering `track` before or after the `success/failure` functions. Thus, I rewrote the above code snippet by changing the title and order.

What do you think about the order?

Reply:
> What do you think about the order?

I'm not very worried about the order, tbh. The only problem I see is that if we have a crash, the track will still be done. But moving to below the `success/failure` won't change much so I can make this change + the renaming. :)

Reply:
> I'm not very worried about the order, tbh. The only problem I see is that if we have a crash, the track will still be done. But moving to below the success/failure won't change much so I can make this change + the renaming. :)

For Android the order matters a bit as there is a bit of computation happening after the API response arrives. Thus, I put it below the corresponding `success/failure` and make sure to name it accordingly (`trackContentPresented`), so that I can be as much precise as possible, contextual and tracking wise.